apis/Google.Cloud.Spanner.Data/docs/index.md

```diff
@@ -128,29 +146,20 @@ Cloud Spanner is fully ACID compliant.
 
 Retries should be implemented at the transaction level (as opposed to individual call level)
 because there is a chance in high stress situations that a Cloud Spanner session (transaction)
-can be canceled due to causing a deadlock. In this case, the entire transaction should be
-retried and run on a new session.
+can be aborted due to causing a deadlock. In this case, the entire transaction should be
+retried. The methods `SpannerConnection.RunWithRetriableTransactionAsync(...)` will take care of
+automatically retrying the entire callback passed as a paremeter using a fresh transaction
+if previous attempts aborted.
 
-Use the extension method IsTransientSpannerFault() on Exception to create your detection
-strategy for the TransientFaultApplication Block.
+- The callback doesn't need to handle the transaction lifecycle.
+- The callback needs to set the transaction received on all the commands it creates.
+- The callback might be called more than once so it needs to be prepared for it.
 
-{{sample:SpannerConnection.SpannerFaultDetectionStrategy}}
-
- You should execute the entire transaction within a RetryPolicy. Generally, we recommend
- using exponential backoff as shown below.
+Please read the [Cloud Spanner user documentation](https://cloud.google.com/spanner/docs/reference/rest/v1/TransactionOptions)
+for details on when and why transactions can be aborted.
 
 {{sample:SpannerConnection.TransactionAsync}}
 
-## TransactionScope support (.NET 4.5+, Single resource only)
```

Comment:
Let's talk offline about this being removed. I think it's reasonable to say we still support TransactionScope, but that it won't retry transactions automatically. If we don't want to support it at all any more, maybe we should think about removing it for the upcoming breaking change.

Reply:
Let's make sure we do that. I suspect we really do need to keep TransactionScope support - which is in netstandard2.0, so we don't need to say ".NET 4.5+" any more - but that we need to emphasize the retry aspect.